test/converters.test.ts

```diff
@@ -358,3 +368,32 @@ test("object converter", async () => {
   const r3 = await converter.convert(null, {});
   expect(r3).toEqual({ value: null });
 });
+
+test("dynamic decimal converter", async () => {
+  const context = { options: { decimalPlaces: 0 } };
+
+  function currency() {
+    return converters.decimal(context => context.options);
+  }
+
+  const M = types.model("M", {
+    foo: types.string
+  });
+
+  const form = new Form(M, {
+    foo: new Field(currency())
+  });
+
+  const o = M.create({ foo: "3" });
+
+  const state = form.state(o, { context: context });
+  const field = state.field("foo");
+
+  await field.setRaw("3.141");
+  expect(field.raw).toEqual("3.141");
+  expect(field.value).toEqual("3"); // conversion error
+  context.options = { decimalPlaces: 3 };
+  await field.setRaw("3.141");
+  expect(field.raw).toEqual("3.141");
+  expect(field.value).toEqual("3.141"); // conversion succeeds
```

Comment:
I expect to see a test where the amount of digits allowed is tightened from 3 to 2, and checking the behavior then. This because I am not entirely sure what the behavior should be. I think what happens is that the raw values remain the same, until you touch the field again by setting the same value on it, and in that case the additional decimal is a conversion error? (I can imagine that's the desired behavior -- you can enter more digits but the system complains if you do).